main.tf

```diff
@@ -127,6 +127,13 @@ resource "aws_instance" "this" {
     }
   }
 
+  dynamic "maintenance_options" {
```

Comment:
Ran across [this PR](https://github.com/terraform-aws-modules/terraform-aws-autoscaling/pull/193) and this implementation made more sense.